modxcms/revolution
# 3.x Fix TV Category not found issue in Template and TV editing panels

### What does it do?
Changed the (grid filtering) category combo configs to:
1. Prevent submission of their values on post
2. Not create a hidden submit field
3. Ignore the initial request value of "undefined" when creating a new Template/TV

The first commit contains the functional changes, while the remaining ones provide code style and quality fixes.

### Why is it needed?
A change somewhere in the current development branch created a condition where form panel pages with both a Category field and a Category grid filter would not save properly. (This is seen only in the current 3.1.0-dev branch.)

**Note:** I added the "Urgent" label to this PR because we'll need it incorporated into 3.1 _before_ its -pl release, as the issue it solves prevents proper operation of the referenced panels.

### How to test

1. Before pulling down this PR, observe that when trying to create a new TV or Template, you are prevented from doing so. The Category field will always have an error, not matter what you do. Also, if you go to the Template Access tab of a TV and filter the grid by Category using the dropdown and then immediately save, note how the Category of the TV itself will have been changed to the one you chose in the grid filter (not what you want).
2. After applying this PR, create and edit a Template and a TV to verify you are able to save and update each as expected, and that filtering on the TV/Template Access grids within each does not affect the saved Element's Category.

### Related issue(s)/PR(s)
None, I discovered this issue when I was working on another PR.
